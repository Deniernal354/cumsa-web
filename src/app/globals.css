@import "tailwindcss";

:root {
  --background: #f8fafc;
  --foreground: #0f172a;
  --surface: #ffffff;
  --muted: #64748b;
  --navy: #2a7487;
  --blue: #5aced8;
  --red: #c73221;
  --header-top: #123a66;
}

body {
  background: var(--background);
  color: var(--foreground);
  font-family: Arial, Helvetica, sans-serif;
}

a {
  text-underline-offset: 4px;
}

.link-cta {
  color: var(--blue);
  font-weight: 700;
  text-decoration: underline;
  text-underline-offset: 3px;
}

.link-cta:hover {
  color: var(--red);
}

.btn-cozy {
  display: inline-flex;
  align-items: center;
  justify-content: center;
  padding: 0.5rem 1rem;
  border-radius: 0.6rem;
  font-weight: 600;
  font-size: 0.9rem;
  transition:
    transform 150ms ease,
    box-shadow 150ms ease,
    opacity 150ms ease;
}

.btn-cozy:hover {
  transform: translateY(-1px);
  box-shadow: 0 6px 16px rgba(0, 0, 0, 0.08);
}

.btn-primary {
  background: var(--blue);
  color: white;
}

.btn-outline {
  background: var(--surface);
  color: var(--foreground);
  border: 1px solid rgba(0, 0, 0, 0.12);
}

:focus-visible {
  outline: 2px solid color-mix(in oklab, var(--blue) 70%, white);
  outline-offset: 2px;
}

.container-base {
  margin-left: auto;
  margin-right: auto;
  max-width: 80rem;
  padding-left: 1rem;
  padding-right: 1rem;
}

@media (min-width: 640px) {
  .container-base {
    padding-left: 1.5rem;
    padding-right: 1.5rem;
  }
}

@media (min-width: 1024px) {
  .container-base {
    padding-left: 2rem;
    padding-right: 2rem;
  }
}

.transition-base {
  transition: all 200ms ease-out;
}

.card {
  background: var(--surface);
  border: 1px solid rgba(0, 0, 0, 0.12);
  border-radius: 0.75rem;
}

.title {
  color: var(--red);
}

.tint-blue {
  filter: invert(28%) sepia(88%) saturate(3612%) hue-rotate(211deg) brightness(93%) contrast(98%);
}

.logo-mask-blue {
  background-color: var(--blue);
  -webkit-mask-image: url("/CU-MSA-LOGO-blank-2.png");
  mask-image: url("/CU-MSA-LOGO-blank-2.png");
  -webkit-mask-repeat: no-repeat;
  mask-repeat: no-repeat;
  -webkit-mask-position: center;
  mask-position: center;
  -webkit-mask-size: contain;
  mask-size: contain;
  display: inline-block;
}
