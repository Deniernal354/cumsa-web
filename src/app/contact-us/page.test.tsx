import { test, expect } from "@jest/globals";
import { render } from "@testing-library/react";
import ContactUsPage from "./page";

test("contact page snapshot", () => {
  const { asFragment } = render(<ContactUsPage />);
  expect(asFragment()).toMatchSnapshot();
});
