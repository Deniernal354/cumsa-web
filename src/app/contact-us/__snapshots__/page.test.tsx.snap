// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`contact page snapshot 1`] = `
<DocumentFragment>
  <div
    class="container-base py-12"
  >
    <h1
      class="text-3xl font-bold"
    >
      Contact Us
    </h1>
    <div
      class="mt-4 grid gap-8 md:grid-cols-2"
    >
      <section>
        <h2
          class="text-xl font-semibold"
        >
          General Inquiries
        </h2>
        <p
          class="mt-2"
        >
          <a
            class="text-[var(--blue)]"
            href="mailto:info@cumsa.ca"
          >
            info@cumsa.ca
          </a>
        </p>
        <div
          class="mt-4 text-gray-700"
        >
          <p>
            Muslim Studentsâ€™ Association
          </p>
          <p>
            Carleton University
          </p>
          <p>
            426 University Centre
          </p>
          <p>
            1125 Colonel By Drive
          </p>
          <p>
            Ottawa, ON K1S 5B6
          </p>
        </div>
        <div
          class="mt-6 text-[var(--blue)]"
          data-testid="stub"
        />
      </section>
      <section>
        <h2
          class="text-xl font-semibold"
        >
          Send us a message
        </h2>
        <form
          class="mt-3 space-y-3"
        >
          <div
            data-testid="stub"
            name="name"
            placeholder="Your name"
            required=""
          />
          <div
            data-testid="stub"
            name="email"
            placeholder="Your email"
            required=""
            type="email"
          />
          <div
            data-testid="stub"
            name="subject"
            placeholder="Subject"
          />
          <textarea
            class="w-full rounded-md border border-black/10 bg-[var(--surface)] px-3 py-2 text-[var(--foreground)] placeholder-[var(--muted)] focus:outline-none focus:ring-2 focus:ring-[var(--blue)]"
            name="message"
            placeholder="Message"
            required=""
            rows="5"
          />
          <div
            class="flex items-center gap-3"
          >
            <div
              data-testid="stub"
              type="submit"
            >
              Send
            </div>
          </div>
        </form>
      </section>
    </div>
  </div>
</DocumentFragment>
`;
